oshi-core/src/main/java/oshi/software/os/OSProcess.java

```diff
@@ -95,6 +106,44 @@
         OTHER
     }
 
+    /**
+     * <p>
+     * Constructor for OSProcess.
+     * </p>
+     */
+    public OSProcess() {
+    }
+
+    /**
+     * <p>
+     * Constructor for OSProcess.
+     * </p>
+     *
+     * @param processID
+     *                 process ID
+     */
+    public OSProcess(int processID) {
+        this.processID = processID;
+        updateAttributes();
+    }
+
+    /**
+     * Updates all process attributes.
+     */
+    public void updateAttributes() {
+        OperatingSystem operatingSystem = getCurrentOperatingSystem();
+        if (operatingSystem != null) {
+            OSProcess process = operatingSystem.getProcess(this.processID);
+            if (process != null) {
+                copyValuesToThisProcess(process);
+            } else {
+                LOG.error("No process found: {}", this.processID);
```

Comment:
This may be a reasonably common return value so error logging isn't appropriate.  I think perhaps we should make this method return boolean, and just do debug level logging here.

Reply:
Hmm. This also introduces the question of what happens when a user attempts to instantiate a process with an invalid pid -- a boolean return from "update" still has to be handled in the constructor. Clearly the stats fail, but since we are instantiating a new object it needs to be non-null and have some way for the user to test whether it was successful.

Reply:
How about static factory method which returns Optional?
`Optional<OSProcess> byPid(int processID);`

Reply:
That's not much different than the current `getProcess(int processID)` except with an optional rather than null.  (And the API for that is fixed so we can't change it.)  

That's not the problem I'm discussing... the problem is that someone writing `new OSProcess(1234)` will definitely get a Process object unless we throw an exception.  We could throw an `InstantiationError`, for example, and require them to handle it.  Or a `MissingResourceException`. Or return a class with the pid and essentially zero data.  Perhaps I should have thought about this dilemma before suggesting the feature!